docs/source/retrieving-results.rst

```diff
@@ -32,6 +32,23 @@ Or find mulitple projects using filters:
     print(project.id, "-", project.name)
 
 
+Retrieving Upload status
+~~~~~~~~~~~~~~~
+Get status of all bulk upload tasks from the last 7 days:
+
+.. code-block:: python
+
+  statuses = client.projects.get_append_status(project_id=project.id)
+
+
+Get status of one upload task
+
+.. code-block:: python
+
+  status = client.projects.get_append_status(project_id=project.id, task_id="18e5b9c4-3498-45e9-a1ac-77659fdd13e1")
```

Comment:
if a user uploads 20k rows with 10k upload tasks (quite a common use case), how would they track the status? polling individual tasks statuses in a loop?

Reply:
When user is asking for status of a full project (without providing id of the task) server is checking all the tasks assigned to the project (in the last 7 days) and returns summary of all tasks, so in example:
all 10k upload tasks (from the last 7 days) are `completed` project status will be `completed` 
if at least one of them will be `in progress` then project status will be `in progress`
if at least one will be `failed` then project status will be `failed`

Reply:
But if user will have 10k uploads in the last 7 days I would suggest to update meerkat API and add there an endpoint to ask for status of multiple tasks, otherwise we will have to request status 10k times from there...

Reply:
Indeed if tracking multiple sequential operations as a whole/project is a common usecase, then we can consider adding such functionality, although if I put my client-of-an-api hat on if I submitted 2(or N) jobs of 10k each and the api I used doesn't seem that I actually appended the 2(N) jobs to the same operation, then I'd find it natural to track the completion of all jobs I submitted.

I've already discussed briefly with Pawel that afaik this is the only operation for which tracking all the operations submitted per project in the last 7 days might make sense. My current impression is that most operations we want to make async actually need the submit-job-track-its-completion mode as opposed to submit-several-jobs-track-completion-off-all-ever-submitted-jobs-for-some-entity(e.g. project).

Reply:
so a user can do something like this after uploading rows to wait until we're finished?
```
while i < max_iter: 
    status_response = client.projects.get_append_status(project_id=project.id)
    if status_response.status == "completed":
        break
    time.sleep(1)
```

Reply:
exactly

Reply:
@pazembrz could we do anything here to make this more convenient for the user, especially since the user also needs to be aware of the rate limit? I was thinking of a function that would poll for our status with some backoff to avoid hitting the rate limit and returning in case of success or raising error if timing out or failing

Reply:
@pazembrz  ping :)

Reply:
@pdeburen  WDYT about this? https://github.com/caplena/caplena-python/pull/18/commits/0d2400c7f2a78c5432e9f09486ab3d164c24057a

Reply:
@pazembrz as far as I understood this makes it such that the API is only hit every 10 seconds right?

Reply:
For requested `project_id` and `task_id` yes. So it will cache for 10s response from `project X` and `task Z`, while still allows to query (also once per 10s) for `project X` and `task Y` or `project A` and `task B`

Reply:
I've extended test to also test this method with task_uuid